crates/subspace-malicious-operator/src/malicious_bundle_producer.rs

```diff
@@ -269,12 +269,21 @@ where
             }
         };
 
-        let maybe_operator_id = self
-            .consensus_client
-            .runtime_api()
-            .operator_id_by_signing_key(consensus_best_hash, signing_key.clone())?;
+        let mut maybe_operator_id = None;
+        for operator_id in current_operators.keys().chain(next_operators.iter()) {
```

Comment:
The checks below handle IDs in current and next, so I think we need to search both.